exporter/kafkaexporter/kafka_exporter_bench_test.go

```diff
@@ -0,0 +1,116 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package kafkaexporter
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/exporter/exportertest"
+	"go.opentelemetry.io/collector/pdata/testdata"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/kafkaexporter/internal/metadata"
+)
+
+var clientType = func() string {
+	if ct := os.Getenv("KAFKA_CLIENT_TYPE"); ct == "" {
+		return ct // Use the value from the environment variable
+	}
+	return clientTypeSarama // Default to Sarama if not set
+}()
+
+// NOTE(marclop) These benchmarks target the default localhost:9092 Kafka broker.
+// You can run them against a real Kafka cluster or a docker container:
+// docker run --rm -d -p 9092:9092 --name broker apache/kafka:4.0.0
+// docker exec --workdir /opt/kafka/bin/ -it broker ./kafka-topics.sh --bootstrap-server localhost:9092 --create --topic otlp_logs
+// docker exec --workdir /opt/kafka/bin/ -it broker ./kafka-topics.sh --bootstrap-server localhost:9092 --create --topic otlp_metrics
+// docker exec --workdir /opt/kafka/bin/ -it broker ./kafka-topics.sh --bootstrap-server localhost:9092 --create --topic otlp_spans
+
+func BenchmarkLogs(b *testing.B) {
+	runBenchmarkLogs(b, clientType)
```

Comment:
If Kafka isn't running locally then all the benchmarks will fail right? Should we add an env var to opt into running them, and skip the benchmarks otherwise?